net/http/balancer.go

```diff
@@ -13,103 +12,109 @@ import (
 	"github.com/trafficstars/registry"
 )
 
-var _balancer balancer
-
-func Init(discovery registry.Discovery) {
-	rand.Seed(time.Now().UnixNano())
-	_balancer = balancer{
-		upstreams: make(map[string]*upstream),
-		discovery: discovery,
-	}
-	_balancer.lookup()
-	go _balancer.supervisor()
-}
+// BalancingStrategy type
+type BalancingStrategy int
 
-type (
-	backend struct {
-		weight      int
-		skipCounter int
-		address     string
-	}
-	backends []*backend
-	upstream struct {
-		index         int
-		gcd           int //greatest common divisor
-		maxWeight     int
-		currentWeight int
-		backends      backends
-	}
+// Balancyng strategy variants
+const (
+	RoundRobinStrategy BalancingStrategy = iota
+	WeightStrategy
 )
 
-func (b *backend) skip() {
-	b.skipCounter = 7
-}
-func (b backends) maxWeight() int {
-	maxWeight := -1
-	for _, backend := range b {
-		if backend.weight > maxWeight {
-			maxWeight = backend.weight
-		}
-	}
-	return maxWeight
-}
+var _balancer balancer
 
-func (b backends) gcd() int {
-	divisor := -1
-	for _, backend := range b {
-		if divisor == -1 {
-			divisor = backend.weight
-		} else {
-			divisor = gcd(divisor, backend.weight)
+// Init balancer based on descovery
+func Init(strategy BalancingStrategy, discovery registry.Discovery, localAddrs ...string) {
+	if len(localAddrs) == 0 || localAddrs[0] == "" {
+		localAddrs, _ = listOfLocalAddresses()
+
+		// all IPv4 addresses on the local machine
+		const allAddress = "0.0.0.0"
+		hasAllAddress := false
+		for _, addr := range localAddrs {
+			if addr == allAddress {
+				hasAllAddress = true
+				break
+			}
+		}
+		if !hasAllAddress {
+			localAddrs = append(localAddrs, allAddress)
```

Comment:
Why is `0.0.0.0` required to be in the slice?

Reply:
Because it not defined in the list of available interfaces

Reply:
No. I meant: why is it required into the slice at all?

Reply:
To make this check more simple and easier to read

```go
	loop:
		for _, bk := range backends {
			for _, addr := range b.localAddrs {
				if addr == bk.hostaddress {
					priorityBackend = bk
					break loop
				}
			}
		}
```

Reply:
Never mind :)